pkg/webhook/cspc.go

```diff
@@ -408,6 +408,23 @@ func validateBlockDevice(bd *openebsapis.BlockDevice, hostName string) error {
 			bd.Labels[types.HostNameLabelKey],
 		)
 	}
+
+	if bd.Labels[types.HostNameLabelKey] != hostName {
+		return errors.Errorf(
+			"block device %s doesn't belongs to node %s",
+			bd.Name,
+			bd.Labels[types.HostNameLabelKey],
+		)
+	}
+
+	if v, found := bd.Labels[types.BlockDeviceTagLabelKey]; found {
```

Comment:
Q: @akhilerm is it possible to have blockdevice tag value as empty?

Reply:
No @mittachaitu . . You should specify the value also. If value is empty the BD will be totally unusable as it can't be claimed by anyone.